lib/javascript/JavascriptParser.js

```diff
@@ -2535,7 +2535,7 @@ class JavascriptParser extends Parser {
 		const renameThis = currentThis ? getVarInfo(currentThis) : null;
 		const varInfoForArgs = options.map(getVarInfo);
 		const wasTopLevel = this.scope.topLevelScope;
-		this.scope.topLevelScope = arrow ? "arrow" : false;
+		this.scope.topLevelScope = wasTopLevel && arrow ? "arrow" : false;
```

Comment:
maybe enum make sense

Reply:
Isn't `true | false | "arrow"` an enum?

Reply:
Technically this is [union type](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#union-types), anyway this is just design question.. In my opinion such design easier to maintain (you don't need to expose type), if you understand what `"arrow"` does. On the other hand enum allows to write self-described code -  e.g. `Scope.ARROW_FUNCTION` is more understandable. Maybe this is not worth it in current case..

Reply:
Definitely makes sense, but this was chosen for backward compat with webpack 4, which had `true | false`. As `"arrow"` is truthy this allows to keep most existing code.